doc/profiling/sonic_bootchart.md

```diff
@@ -0,0 +1,202 @@
+
+<p align=center>
+<img src="img/bootchart.svg" alt="Figure 1. Bootchart example">
+</p>
+
+### Requirements
+
+- Integrate systemd-bootchart with SONiC OS
+- systemd-bootchart is by default installed in the system
+- systemd-bootchart is by default disabled and user need to enable it via CLI
+- SONiC CLI tool to interact with systemd-bootchart
+- Support commands to enable/disable systemd-bootchart
+- Configure the amount of samples to collect and frequency
+- Displaying systemd-bootchart configuration and generated SVG plots
+
+### Architecture Design
+
+N/A
+
+### High-Level Design
+
+SONiC build system includes a new build-time flag to INCLUDE_BOOTCHART (y/n). When this flag is set a *systemd-bootchart* debian package is installed in SONiC host from upstream debian repositories (Installed size 128 KB).
+
+Example INCLUDE_BOOTCHART flag usage:
+
+Include bootchart:
+```
+make INCLUDE_BOOTCHART=y target/sonic-mellanox.bin
+```
+
+Do not include bootchart:
+```
+make INCLUDE_BOOTCHART=n target/sonic-mellanox.bin
+```
+
+SONiC provides a default configuration for bootchart located at /etc/systemd/bootchart.conf:
+
+```ini
+[Bootchart]
+Samples=4500
+Frequency=25
+```
+
+This configuration means that samples are collected for 3 minutes of uptime with a frequency of 25 samples per second.
+
+SONiC utilities is extendeded with a new script under *scripts/* to include a new utility *sonic-bootchart*.
+
+### SAI API
+
+N/A
+
+### Configuration and management
+
+*sonic-bootchart* is a standalone utility, like *sonic-kdump-config*, *sonic-installer*, etc.
+
+Command line interface:
+
+```
+admin@sonic:~$ sudo sonic-bootchart
+Usage: sonic-bootchart [OPTIONS] COMMAND [ARGS]...
+
+  Main CLI group
+
+Options:
+  --help  Show this message and exit.
+
+Commands:
+  config   Configure bootchart NOTE: This command requires elevated (root)...
+  disable  Disable bootchart NOTE: This command requires elevated (root)...
+  enable   Enable bootchart NOTE: This command requires elevated (root)...
+  show     Display bootchart configuration
+```
+
+Enable/Disable bootchart:
+
+```
+admin@sonic:~$ sudo sonic-bootchart enable
+Running command: systemctl enable systemd-bootchart
+admin@sonic:~$ sudo sonic-bootchart disable
+Running command: systemctl disable systemd-bootchart
+```
+
+In case image is built without bootchart included (INCLUDE_BOOTCHART=n) sonic-bootchart will print an error:
+
+```
+admin@sonic:~$ sudo sonic-bootchart enable
+systemd-bootchart is not installed
+```
+
+Once bootchart is enabled a systemd-bootchart.service is enabled in systemd which starts a daemon early at boot that collects samples.
+This setting is permanent, meaning, after it is enabled, bootchart will always run at boot until disabled, *config save*, *config reload* do not affect this setting.
+
+
+Update configuration example:
+
+```
+admin@sonic:~$ sudo sonic-bootchart config --samples 500 --frequency 10
```

Comment:
Understand bootchart's parameters are sample and frequency, as cli wrapper, I suggest:
1. Instead of sample counts, use time span as parameter, wrapper calculate the sample count = time span x frequency
2. having meaningful default value for time span and frequency.